src/controllers/UserController.ts

```diff
@@ -0,0 +1,59 @@
+/**
+ * Italia PagoPA Proxy
+ * Cittadinanza Digitale PagoPA services
+ */
+
+import * as express from "express";
+import * as Mocked from "../MockedData";
+import { CreditCard } from "../types/CreditCard";
+import { PaymentMethodEnum } from "../types/PaymentMethod";
+import { extractUserFromRequest } from "../types/User";
+import { ControllerUtils } from "../utils/ControllerUtils";
+
+// User Controller
+export class UserController {
+  public getWallets(req: express.Request, res: express.Response): void {
+    const errorOrUser = extractUserFromRequest(req);
+    if (errorOrUser.isLeft()) {
+      ControllerUtils.setErrorResponse(res, errorOrUser.value);
+    } else {
+      ControllerUtils.setSuccessResponse(res, { wallet: Mocked.wallet });
+    }
+  }
+
+  public getCreditCards(req: express.Request, res: express.Response): void {
+    const errorOrUser = extractUserFromRequest(req);
+    if (errorOrUser.isLeft()) {
+      ControllerUtils.setErrorResponse(res, errorOrUser.value);
+    } else {
+      if (req.params.cardId === undefined) {
+        ControllerUtils.setSuccessResponse(res, {
+          credit_cards: Mocked.wallet
+            .filter(method => method.type === PaymentMethodEnum.CREDIT_CARD)
```

Comment:
Since `CreditCard ` type is defined via [io-ts](https://github.com/gcanti/io-ts), you get `CreditCard.is(...)` for free -  `CreditCard.is(...)` is a [type guard](https://basarat.gitbooks.io/typescript/docs/types/typeGuard.html) for `CreditCard `, using it will avoid casting:

```ts
credit_cards: Mocked.wallet.filter(CreditCard.is) // CreditCard[]
```